openscap/src/main/scala/com/normation/plugins/openscappolicies/extension/OpenScapNodeDetailsExtension.scala

```diff
@@ -129,13 +145,46 @@ class OpenScapNodeDetailsExtension(
 
   }
 
-  def frameContent(nodeId: NodeId, hostname: String): CssSel = {
+  def frameContent(file: File, report: OpenScapReport): CssSel = {
+    import report.*
+    val sanitizedReport = ReportSanitizer.sanitizeHTMLReport(report.content)
+    val baseName        = s"OpenSCAP report for ${hostname} (${nodeId.value})"
+
+    "iframe [srcdoc]" #> sanitizedReport &
+    ".sanitized" #> downloadReportButton(
+      s"${baseName} - sanitized.html",
+      sanitizedReport.getBytes(StandardCharsets.UTF_8),
+      sanitized = true
+    ) &
+    ".original" #> downloadReportButton(
+      s"${baseName}.html",
+      report.content.getBytes(StandardCharsets.UTF_8),
+      sanitized = false
+    )
+
+  }
 
-    "iframe [src]" #> s"/secure/api/openscap/sanitized/${nodeId.value}" &
-    ".sanitized [href]" #> s"/secure/api/openscap/sanitized/${nodeId.value}" &
-    ".original [href]" #> s"/secure/api/openscap/report/${nodeId.value}" &
-    ".original [download]" #> s"OpenSCAP report for ${hostname} (${nodeId.value}).html"
+  private def downloadReportButton(fileName: String, bytes: Array[Byte], sanitized: Boolean) = {
+    val base64   = Base64.getEncoder().encodeToString(bytes)
+    val download = Run(
+      s"""saveByteArray("${fileName}", "application/zip", base64ToArrayBuffer("${base64}"));"""
+    )
+
+    val btn = if (sanitized) {
+      SHtml.ajaxButton(
+        <span><i class="fa fa-download text-light"></i>Download report</span>,
```

Comment:
We don't want a download here, we want to open a tab with the report. That allows to see it in full page, with link navigation available, which is mandatory to be able to see anything.